core/src/main/scala/sttp/tapir/DecodeResult.scala

```diff
@@ -35,6 +35,7 @@ object DecodeResult {
     }
   }
   case class Mismatch(expected: String, actual: String) extends Failure
+  case class BodyTooLarge(maxBytes: Long) extends Failure
```

Comment:
I'd try to avoid creating new `Failure` cases. Not sure if that's even bin-compatible. Maybe we can use `Error(contentReadSoFar, StreamMaxLengthExceededException)` ?

Reply:
`contentReadSoFar` may be problematic in case of binary data. Also, the limit may be pretty high, and we don't want to put large content in the error message, right? I'll check whether an empty string here gives anything acceptable.

Reply:
If we don't have the data in a`String` format I'd just leave this empty